File: server/OurCity.Api/Infrastructure/Database/Post.cs
using OurCity.Api.Common.Enum;

namespace OurCity.Api.Infrastructure.Database;

public class Post
{
    public Guid Id { get; set; }
    
    public Guid AuthorId { get; set; }  

    public required string Title { get; set; }

    public required string Description { get; set; }

    public string? Location { get; set; }

    public PostVisibility Visisbility { get; set; } = PostVisibility.Published;

    public bool IsDeleted { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    // Navigation Properties
    public User? Author { get; set; }

    public List<Comment> Comments { get; set; } = new(); 

    public List<Tag> Tags { get; set; } = new();

    public List<PostVote> Votes { get; set; } = new();
}

File: webapp/src/models/post.ts
/*
  /webapp/src/models/post.ts

  This file contains the User domain model, which represents the User entity in the application.
*/
import type { Tag } from "./tag";
import { PostVisibility, VoteType } from "../types/enums";

// Post model representing the Post entity in the application
export interface Post {
  id: string;
  authorId: string;
  title: string;
  description: string;
  authorName?: string;
  location?: string;
  latitude?: number;
  longitude?: number;
  upvoteCount: number;
  downvoteCount: number;
  voteCount: number; // total votes (upvotes - downvotes)
  commentCount: number;
  visibility: PostVisibility; // PUBLISHED | HIDDEN
  tags: Tag[];
  isDeleted: boolean;
  canMutate: boolean; // authorization flag from backend
  voteStatus: VoteType; // UPVOTE = 1| DOWNVOTE = -1| NONE = 0
  isBookmarked: boolean;
  isReported: boolean;
  createdAt: Date;
  updatedAt: Date;
}
